src/v/cloud_storage/remote_segment.cc

```diff
@@ -358,6 +364,16 @@ ss::future<bool> remote_segment::do_materialize_segment() {
           "re-hydrated, {} waiter are pending",
           _path,
           _wait_list.size());
+
+        // If we got here, the cache is in a stressed state: it has
+        // evicted an object that we probably only just promoted.  We can
+        // live-lock if many readers are all trying to promote their objects
+        // concurrently and none of them is getting all their objects in
+        // the cache at the same time: reduce chance of this by backing off.
+        // TODO: this should be a sleep_abortable, but remote_segment does not
+        // have an abort source.
+        co_await ss::sleep(_cache_backoff_jitter.next_duration());
```

Comment:
might be useful to feed an abort source into this

EDIT: ah i see that mentioned in the comment